### ISSUE TITLE
Case sensitive titles for the keywords

### ISSUE THREAD

**Issue by geoffroy-noel-ddh** [stephenmcd/mezzanine#647]
Hi,

This is a feature request for adding support for case sensitive keywords. When I create a keyword using the API (Keyword model and Keywords field) it is possible to have a title with uppercase letters.

When I go to the Admin interface to edit the keywords of one of my own models, I can see this keyword with uppercase (e.g. 'Test') listed under the Keywords widget. When I click it and then save a new keyword record is created as a lower case version of my uppercase keyword (e.g. 'test') and assigned it to the model. I end up with two keywords (Test and test) and only the lowercase one is assigned.

Basically keywords_field.js will call generic/views.py:admin_keywords_submit() which in turn converts all the keywords to lowercase. If I remove the call to lower() it seems to fix the issue.

I'm just curious about the reason for converting to lower case before doing a lookup in the keywords table?

Thanks,

Jeff
